wire-ios-data-model/Source/Model/UserClient/UserClient.swift

```diff
@@ -527,125 +507,6 @@ public extension UserClient {
         self.markedToDelete = true
         self.setLocallyModifiedKeys(Set(arrayLiteral: ZMUserClientMarkedToDeleteKey))
     }
-
-    @available(*, deprecated)
-    func markForFetchingPreKeys() {
-        self.fetchFingerprintOrPrekeys()
-    }
-
-    @objc func fetchFingerprintOrPrekeys() {
-        guard
-            self.fingerprint == .none,
-            let syncMOC = self.managedObjectContext?.zm_sync
-        else {
-            return
-        }
-
-        if self.objectID.isTemporaryID {
-            do {
-                try syncMOC.obtainPermanentIDs(for: [self])
-            } catch {
-                fatal("Error obtaining permanent id for client")
-            }
-        }
-
-        let selfObjectID = self.objectID
-
-        syncMOC.performGroupedBlock({ [unowned syncMOC] () -> Void in
-            guard
-                let obj = try? syncMOC.existingObject(with: selfObjectID),
-                let syncClient = obj as? UserClient,
-                let syncSelfClient = ZMUser.selfUser(in: syncMOC).selfClient()
-            else {
-                return
-            }
-
-            if syncSelfClient == syncClient {
-                syncClient.fingerprint = syncClient.localFingerprint()
-                syncMOC.saveOrRollback()
-            } else {
-                if !syncClient.hasSessionWithSelfClient {
-                    syncSelfClient.missesClient(syncClient)
-                    syncSelfClient.setLocallyModifiedKeys(Set(arrayLiteral: ZMUserClientMissingKey))
```

Comment:
I assume it is what the SessionEstablisher replaces in establishSession. @typfel ?

Reply:
Exactly!